File: majorProject/src/main/java/com/aocc/framework/Input.java
package com.aocc.framework;

import java.util.List;

//EXCEPT WHERE NOTED, THE FOLLOWING CODE IS SOURCED FROM THE KILOBOLT ANDROID FRAMEWORK

public interface Input {
    
    public static class TouchEvent {
        public static final int TOUCH_DOWN = 0;
        public static final int TOUCH_UP = 1;
        public static final int TOUCH_DRAGGED = 2;
        public static final int TOUCH_HOLD = 3;

        public int type;
        public int x, y;
        public int pointer;


    }

    public boolean isTouchDown(int pointer);

    public int getTouchX(int pointer);

    public int getTouchY(int pointer);

    public List<TouchEvent> getTouchEvents();
    
    public float getRotationX();

    public float getRotationY();

    public float getRotationZ();
}
 

File: majorProject/src/main/kotlin/com/aocc/framework/Input.kt
package com.aocc.framework

//EXCEPT WHERE NOTED, THE FOLLOWING CODE IS SOURCED FROM THE KILOBOLT ANDROID FRAMEWORK

interface Input {

    class TouchEvent {
        @JvmField
        var type: Int = 0
        @JvmField
        var x: Int = 0
        @JvmField
        var y: Int = 0
        @JvmField
        var pointer: Int = 0

        companion object {
            const val TOUCH_DOWN = 0
            const val TOUCH_UP = 1
            const val TOUCH_DRAGGED = 2
            const val TOUCH_HOLD = 3
        }
    }

    fun isTouchDown(pointer: Int): Boolean

    fun getTouchX(pointer: Int): Int

    fun getTouchY(pointer: Int): Int

    val touchEvents: List<TouchEvent>

    fun getRotationX(): Float

    fun getRotationY(): Float

    fun getRotationZ(): Float
}
